alchemy/src/vercel/project.ts

```diff
@@ -0,0 +1,418 @@
+import type { Context } from "../context.js";
+import { Resource } from "../resource.js";
+import type { Secret } from "../secret.js";
+import { createVercelApi } from "./api.js";
+
+/**
+ * Properties for creating or updating a Project
+ */
+export interface ProjectProps {
+  /**
+   * The desired name for the project
+   *
+   * Maximum length: `100`
+   * Example: `a-project-name`
+   */
+  name: string;
+
+  /**
+   * Opt-in to preview toolbar on the project level
+   */
+  enablePreviewFeedback?: boolean;
+
+  /**
+   * Opt-in to production toolbar on the project level
+   */
+  enableProductionFeedback?: boolean;
+
+  /**
+   * The build command for this project. When `null` is used this value will be automatically detected
+   */
+  buildCommand?: string;
+
+  commandForIgnoringBuildStep?: string;
+
+  /**
+   * The dev command for this project. When `null` is used this value will be automatically detected
+   */
+  devCommand?: string;
+
+  /**
+   * Collection of ENV Variables the Project will use
+   */
+  environmentVariables?: Array<
+    {
+      /**
+       * The key of the environment variable
+       */
+      key: string;
+
+      /**
+       * The target environment
+       */
+      target: ("production" | "preview" | "development")[];
```

Comment:
Should this inner type be pulled out and exposed as `ProjectEnvironmentVariable`

Reply:
Agreed!